### DISCUSSION TITLE
Some text characters in the content stream are present in a different format when the content stream is extracted using doc.xref_stream().decode()

### DISCUSSION THREAD

**Discussion by meghanaviyyapu** [pymupdf/PyMuPDF#1368] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/36301306/140279903-a40ea0b8-d3d7-44cc-ae43-37ae2d2cec59.png)
Instead of "fi" I observed /037.Can you please let me know why some characters are being present in different format in the stream.

**Comment by JorjMcKie** [comment, marked as answer]
Some fonts support so-called ligatures. These are single glyphs that represent more than one character. MuPDF supports these 7:
```python
>>> for i in range(7):
	print(chr(0xfb00 + i))

	
ﬀ
ﬁ
ﬂ
ﬃ
ﬄ
ﬅ
ﬆ
>>> 
```
So in your font, 0xFB01 = "fi" is represented by the glyph id 0o37 (as an octal number).

**Reply by meghanaviyyapu** [reply]
Okay Thank you for the explanation. Can you please let me know if there are any glyphs apart from the above seven.

**Comment by JorjMcKie** [comment]
Strictly speaking, many languages have ligatures as part of their normal alphabet like the German "umlauts" ä. ö, ü, etc. which came from ae, oe, ue, or ß => ss, or think of the many special Scandinavian letters.
The Devanagari script used in the Indian subcontinent has lots and lots of them.
So there is no final answer to your question but the above seven or even just the first 6 are the usual ones in the extended Latin alphabet.

**Reply by meghanaviyyapu** [reply]
Okay thank you .

**Reply by JorjMcKie** [reply]
what most people do not know: "w" originated as a ligature from "v + v", and "&" is a ligature made from "et" (Latin for "and").

**Reply by meghanaviyyapu** [reply]
Using doc.xref_set_key(), I am trying to add tags to PDF but can you let me know what value should be provided to /K for text elements like links, headings.
![image](https://user-images.githubusercontent.com/36301306/140308075-ad4bad66-e7d5-40a2-b148-11c277d67512.png)

**Reply by JorjMcKie** [reply]
Sorry, I have no experience in this area.
